jobwatcher/plugins/utils.py

```diff
@@ -0,0 +1,39 @@
+import shlex
+import subprocess as sub
+import os
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def run_command(command, env):
+    _command = shlex.split(command)
+    try:
+        DEV_NULL = open(os.devnull, "rb")
+        env.update(os.environ.copy())
+        process = sub.Popen(_command, env=env, stdout=sub.PIPE, stderr=sub.STDOUT, stdin=DEV_NULL)
+        _output = process.communicate()[0]
+        return _output
+    except sub.CalledProcessError:
+        log.error("Failed to run %s\n" % _command)
+        exit(1)
+    finally:
+        DEV_NULL.close()
+
+
+def get_optimal_nodes(nodes_requested, slots_requested, instance_properties):
```

Comment:
I think this is going to under-scale in few scenarios that we may want to understand if we want to address, e.g.  in an env with 4 procs per node and 3 jobs submitted with the following requiremetns
job1: 4 nodes/2 proc per node
job2: 1 node / 4 proc
job3: 1 node/ 4 proc
Total # slots_required = 16 - so according to this approach the first 4 nodes would be enough when instead 6 nodes woudl be required.
The proposed 'filling jars' idea was coping with this, let's talk about this if it make sense

Reply:
Changed the design